src/error.rs

```diff
@@ -94,7 +94,7 @@ pub enum RoutingError {
     /// Serialisation Error
     SerialisationError(serialisation::SerialisationError),
     /// Asymmetric Decryption Failure
-    AsymmetricDecryptionFailure,
+    AsymmetricDecryptionFailure(safe_crypto::Error),
```

Comment:
Is implementing `From` also not better ? That way you probably don't have to do `map_err`; simply `?` would convert it for you.